pallets/committee/src/types.rs

```diff
@@ -66,33 +109,61 @@ impl<AccountId: Default + PartialEq, BlockNumber: Default> VoteAggregate<Account
     }
 
     // This does not check if a vote is a duplicate, This must be done before calling this function
-    pub fn cast_vote(&mut self, voter: AccountId, vote: &Vote) {
-        match vote {
-            Vote::Aye => self.ayes.push(voter),
-            Vote::Nay => self.nays.push(voter),
-            Vote::Abstain => self.abstentions.push(voter),
-        }
+    pub fn cast_vote(&mut self, vote: MemberVote<AccountId>) {
+        self.votes.push(vote)
     }
 
     pub fn remove_voters(&mut self, voters: &[AccountId]) {
-        self.ayes.retain(|x| !voters.contains(x));
-        self.nays.retain(|x| !voters.contains(x));
-        self.abstentions.retain(|x| !voters.contains(x));
+        self.votes
+            .retain(|x| !voters.contains(&x.member.account_id));
     }
 
     pub fn has_voted(&self, voter: &AccountId) -> bool {
-        self.ayes.contains(voter) | self.nays.contains(voter) | self.abstentions.contains(voter)
+        self.votes
+            .iter()
+            .filter(|x| &x.member.account_id == voter)
+            .count()
+            > 0
     }
 
-    pub fn participants(&self) -> usize {
-        self.ayes.len() + self.nays.len() + self.abstentions.len()
+    /// produce a tuple of the vote totals: (ayes, nays, abstentions)
+    /// Can optionally filter by membership type to only tally council or constituent votes
+    pub fn tally(&self, member_type: &Option<MemberType>) -> (usize, usize, usize) {
```

Comment:
With optional input arguments, I'm usually not sure which variant is more appropriate:

`&Option<T>` or `Option<&T>`, while you can turn `Option<T>` easily into the later with `as_ref`, turning a `&T` into `&Option<T>` would require taking ownership if I'm not mistaken.

However I don't think that this decision has any impact here.

Reply:
Yeah I'm also not sure in this case. I switched to `Option<&T>`. I think passing an option that holds a reference to T makes more sense than passing a reference to an option that owns T